src/datachain/lib/dc.py

```diff
@@ -290,18 +276,43 @@ def column(self, name: str) -> Column:
     def c(self, column: Union[str, Column]) -> Column:
         """Returns Column instance attached to the current chain."""
         c = self.column(column) if isinstance(column, str) else self.column(column.name)
-        c.table = self.table
+        c.table = self._query.table
         return c
 
+    @property
+    def session(self) -> Session:
+        """Session of the chain."""
+        return self._query.session
+
+    @property
+    def name(self) -> Optional[str]:
+        """Name of the chain."""
+        return self._query.name
+
+    @property
+    def version(self) -> Optional[int]:
+        """Version of the chain."""
+        return self._query.version
+
+    def __or__(self, other: "Self") -> "Self":
+        """Return `self.union(other)`."""
+        return self.union(other)
+
     def print_schema(self) -> None:
         """Print schema of the chain."""
         self._effective_signals_schema.print_tree()
 
-    def clone(self, new_table: bool = True) -> "Self":
+    def clone(self, *, query=None, new_table: bool = True) -> "Self":
```

Comment:
Since `clone()` is part of the public API, I'm actually not sure it's a good idea to coopt it like this.

Reply:
I think that's fine for now